lib/trailblazer/endpoint.rb

```diff
@@ -1,53 +1,117 @@
-        resolve: ->(result) { result })
-    )
-
-    # `call`s the operation.
-    def self.call(operation_class, handlers, *args, &block)
-      result = operation_class.(*args)
-      new.(result, handlers, &block)
+    DEFAULT_MATCHERS = {
+      created: {
+        rule: ->(result) { result.success? && result["model.action"] == :new },
+        resolve: lambda do |result, representer|
+          { "data": representer.new(result[:model]), "status": :created }
+        end
+      },
+      deleted: {
+        rule: ->(result) { result.success? && result["model.action"] == :destroy },
+        resolve: lambda do |result, _representer|
+          { "data": { id: result[:model].id }, "status": :ok }
+        end
+      },
+      found: {
+        rule: ->(result) { result.success? && result["model.action"] == :find_by },
+        resolve: lambda do |result, representer|
+          { "data": representer.new(result[:model]), "status": :ok }
+        end
+      },
+      success: {
+        rule: ->(result) { result.success? },
+        resolve: lambda do |result, representer|
+          data = if representer
+                   representer.new(result[:results])
+                 else
+                   result[:results]
+                 end
+          { "data": data, "status": :ok }
+        end
+      },
+      unauthenticated: {
+        rule: ->(result) { result.policy_error? },
+        resolve: ->(_result, _representer) { { "data": {}, "status": :unauthorized } }
+      },
+      not_found: {
+        rule: ->(result) { result.failure? && result["result.model"]&.failure? },
+        resolve: lambda do |result, _representer|
+          {
+            "data": { errors: result["result.model.errors"] },
+            "status": :unprocessable_entity
+          }
+        end
+      },
+      invalid: {
+        rule: ->(result) { result.failure? },
+        resolve: lambda do |result, _representer|
+          {
+            "data": { errors: result.errors || result[:errors] },
+            "status": :unprocessable_entity
+          }
+        end
+      },
+      fallback: {
+        rule: ->(_result) { true },
+        resolve: lambda do |_result, _representer|
+          { "data": { errors: "Can't process the result" },
+            "status": :unprocessable_entity }
+        end
+      }
+    }.freeze
+
+    # NOTE: options expects a TRB Operation result
+    # it might have a representer, else will assume the default name
+    def self.call(operation_result, representer_class = nil, overrides = {})
+      endpoint_opts = { result: operation_result, representer: representer_class }
+      new.(endpoint_opts, overrides)
     end
 
-    def call(result, handlers=nil, &block)
-      matcher.(result, &block) and return if block_given? # evaluate user blocks first.
-      matcher.(result, &handlers)     # then, generic Rails handlers in controller context.
+    def call(options, overrides)
```

Comment:
In examples and tests you have one more argument

Reply:
Sorry, the tests are not yet updated to match the final implementation. They were following my mid way solution. There is definitely a little bit of work to be done here for the tests and readme. Was trying to get the general feeling for the actual solution.